src/DssEmergencySpell.sol

```diff
@@ -0,0 +1,135 @@
+    bytes public constant sig = abi.encodeWithSelector(DssAction.execute.selector);
+    /// @dev Emergency spells should not expire.
+    uint256 public constant expiration = type(uint256).max;
+    // @dev Office Hours is always `false` for emergency spells.
+    bool public constant officeHours = false;
+    // @dev `action` is expected to return a valid address.
+    //      We also implement the `DssAction` interface in this contract.
+    address public immutable action = address(this);
+    // @dev An emergency spell can be cast as soon as it is deployed.
+    //      Notice that cast is always a no-op.
+    uint256 internal immutable _nextCastTime = block.timestamp;
+
+    /**
+     * @dev In regular spells, `tag` is an immutable variable with the code hash of the spell action.
+     *      It specifically uses a separate contract for spell action because `tag` is immutable and the code hash of
+     *      the contract being initialized is not accessible in the constructor.
+     *      Since we do not have a separate contract for actions in Emergency Spells, `tag` has to be turned into a
+     *      getter function instead of an immutable variable.
+     * @return The contract codehash.
+     */
+    function tag() external view returns (bytes32) {
+        return address(this).codehash;
+    }
+
+    /**
+     * @notice Triggers the emergency actions of the spell.
+     * @dev Emergency spells are triggered when scheduled.
+     *      This function maintains the name for compatibility with regular spells, however nothing is actually being
+     *      scheduled. Emergency spells take affect immediately, so there is no need to call `pause.plot()`.
+     */
+    function schedule() external {
+        _emergencyActions();
+    }
+
+    /**
+     * @notice Implements the emergency actions to be triggered by the spell.
+     */
+    function _emergencyActions() internal virtual;
+
+    /**
+     * @notice Returns `_nextCastTime`.
+     * @dev this function exists only to keep interface compatibility with regular spells.
+     */
+    function nextCastTime() external view returns (uint256 castTime) {
+        return _nextCastTime;
+    }
+
+    /**
+     * @notice No-op.
+     * @dev this function exists only to keep interface compatibility with regular spells.
+     */
+    function cast() external {}
+
+    /**
+     * @notice No-op.
+     * @dev this function exists only to keep interface compatibility with regular spells.
+     */
+    function execute() external {}
+
+    /**
+     * @notice No-op.
+     * @dev this function exists only to keep interface compatibility with regular spells.
+     */
+    function actions() external {}
+
+    /**
+     * @notice Returns `nextCastTime`, regardless of the input parameter.
+     * @dev this function exists only to keep interface compatibility with regular spells.
+     */
+    function nextCastTime(uint256) external view returns (uint256 castTime) {
+        return _nextCastTime;
+    }
```

Comment:
This function is defined two times

Reply:
It's not. If you notice, there are 2 different signatures:
- `function nextCastTime() external`, to comply with [`DssExec`](https://github.com/makerdao/dss-exec-lib/blob/69b658f35d8618272cd139dfc18c5713caf6b96b/src/DssExec.sol#L60)
- `function nextCastTime(uint256) external`, to comply with [`DssAction`](https://github.com/makerdao/dss-exec-lib/blob/69b658f35d8618272cd139dfc18c5713caf6b96b/src/DssAction.sol#L62)